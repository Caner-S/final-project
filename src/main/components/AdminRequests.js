import React from 'react';
import TableContainer from "@material-ui/core/TableContainer";
import Paper from "@material-ui/core/Paper";
import Table from "@material-ui/core/Table";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import TableCell from "@material-ui/core/TableCell";
import TableBody from "@material-ui/core/TableBody";
import "./Spaces.css";
import {getAllPendingRequests, updateStatus} from "../dao/RequestDao";
import { makeBooking} from "../dao/BookingDao";
import Confirm from "./Confirm";
import firebase from "../../config/firebase";

class AdminRequests extends React.Component {
    constructor() {
        super();
        this.state = {
            requests: [],
        };

    }

    componentDidMount() {
        let currentComponent = this;
        if(firebase.auth().currentUser){
        getAllPendingRequests(firebase.auth().currentUser.uid).then(function(querySnapshot) {
            querySnapshot.forEach(function(doc) {
                let updatedData = querySnapshot.docs.map(doc => doc)
                currentComponent.setState({ requests: updatedData })

            })
        })
        }



    }


    approveRequest(request){
        //set status to approved
        makeBooking(request.data().spaceId, request.data().userId, request.data().arrivalDate, request.data().departureDate);
        updateStatus(request.id, "Approved");
    }

    cancelRequest(request){
        //set status to denied
        updateStatus(request.id, "Denied");

    }



    render() {
        let {requests} = this.state;
        return (
            <div className="BookingsBorder">
                <TableContainer component={Paper} className="container">
                    <Table aria-label="simple table">
                        <TableHead>
                            <TableRow>
                                <TableCell>User&nbsp;Id</TableCell>
                                <TableCell align="right">Space&nbsp;Id</TableCell>
                                <TableCell align="right">Arrival Time</TableCell>
                                <TableCell align="right">Departure Time</TableCell>
                                <TableCell align="right">Business Case</TableCell>
                            </TableRow>
                        </TableHead>
                        <TableBody>
                            {requests.map(request => (
                                <TableRow key={request.id} >
                                    <TableCell component="th" scope="row">
                                        {request.data().userId}
                                    </TableCell>
                                    <TableCell align="right">{request.data().spaceId}</TableCell>
                                    <TableCell align="right">{new Date(request.data().arrivalDate).toLocaleString()}</TableCell>
                                    <TableCell align="right">{new Date(request.data().departureDate).toLocaleString()}</TableCell>
                                    <TableCell align="right">{request.data().businessCase}</TableCell>
                                    <TableCell align="right">
                                        <Confirm buttonText="Approve" title={"Approve a request"} description={"Are you sure you want to approve request "+request.data().spaceId+"?"} onAccept={() => { this.approveRequest(request)}} />

                                    </TableCell>
                                    <TableCell align="right">
                                        <Confirm buttonText="Deny" title={"Deny a request"} description={"Are you sure you want to deny request "+request.data().spaceId+"?"} onAccept={() => { this.cancelRequest(request)}} />

                                    </TableCell>
                                </TableRow>

                            ))}
                        </TableBody>
                    </Table>
                </TableContainer>
            </div>
        );
    }
}
export default AdminRequests;