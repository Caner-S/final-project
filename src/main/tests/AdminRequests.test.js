import React from 'react';
import renderer from 'react-test-renderer';

import AdminRequests from "../components/AdminRequests";

it('renders correctly when there are no items', () => {
    const tree = renderer.create(<AdminRequests />).toJSON();
    expect(tree).toMatchSnapshot();
});