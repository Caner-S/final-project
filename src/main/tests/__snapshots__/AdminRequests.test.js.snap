// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="BookingsBorder"
>
  <div
    className="MuiPaper-root MuiTableContainer-root container MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      aria-label="simple table"
      className="MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            User Id
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Space Id
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Arrival Time
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Departure Time
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Business Case
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      />
    </table>
  </div>
</div>
`;
